### PR TITLE
Re-implement Command#exec()

### PR DESCRIPTION
**Pull request by youpong**
Re-implemented Command#exec() method with exec(3) instead of popen(3).
This change is a pull-request to fulfil issue #153.

Due to this change, pipes are not available. Specifically, commands such as the following cannot be executed

```c++
Command::exec("lspci | grep VGA");
```

For this reason, the implementation of getGPU() was changed.

Command#getExitCode() returns 127 and 126 if the executable does not exist or permission to execute is not given, respectively.

Added Command#getErrorOutput()

Added test/seafood.sh to test Command#exec()

### CODE REVIEW

**Comment by TanmayPatil105** [comment]
Apologies!\
My laptop has been malfunctioning for the past few days. I'll try to review as soon as possible.

**Comment by youpong** [comment]
That's tough. I hope your computer gets fixed soon.

File: src/fetch.h [hunk 1]
```diff
@@ -253,31 +296,125 @@ class Command
...
+        FILE *err = fdopen(err_fd[0], "r");
+        if (err == NULL)
         {
-            throw runtime_error("popen failed: \""s + cmd + "\""s);
+            throw runtime_error("fdopen failed");
         }
 
         int c;
-        while ((c = fgetc(pipe)) != EOF)
+        while ((c = fgetc(out)) != EOF)
```
Comment (TanmayPatil105) [inline src/fetch.h:372]: Reading output character-by-character can be costly if it's too large. Can we use `getline()` here instead?

File: src/fetch.h [hunk 2]
```diff
@@ -253,31 +296,125 @@ class Command
...
                 result->lines += 1;
             }
             result->output += c;
         }
-        // Don't concise below 2 lines. It must be assigned to a variable for
-        // macOS.
-        int n = pclose(pipe);
-        result->exit_code = WEXITSTATUS(n);
+
+        while ((c = fgetc(err)) != EOF)
```
Comment (TanmayPatil105) [inline src/fetch.h:381]: Same as above!

File: src/fetch.h [hunk 3]
```diff
@@ -253,31 +296,125 @@ class Command
...
+            throw runtime_error("fclose failed: " + string(strerror(errno)));
+        }
+
+        if (fclose(err) == EOF)
+        {
+            throw runtime_error("fclose failed: " + string(strerror(errno)));
+        }
+
+        int status;
+        waitpid(pid, &status, 0);
```
Comment (TanmayPatil105) [inline src/fetch.h:397]: I thought of calling `waitpid` before `fdopen`s and `read`s but that would just slow down the execution.\
This is ~correct~ better 👍🏻 .

**Comment by TanmayPatil105** [comment]
Thanks!

**Pull request merged**